client/layout/masterbar/logged-in.jsx

```diff
@@ -47,6 +51,23 @@ class MasterbarLoggedIn extends React.Component {
 	clickMySites = () => {
 		this.props.recordTracksEvent( 'calypso_masterbar_my_sites_clicked' );
 		this.props.setNextLayoutFocus( 'sidebar' );
+
+		/**
+		 * Site Migration: Reset a failed migration when clicking on My Sites
+		 *
+		 * If the site migration has failed, clicking on My Sites sends the customer in a loop
+		 * until they click the Try Again button on the migration screen.
+		 *
+		 * This code makes it possible to reset the failed migration state when clicking My Sites too.
+		 */
+		if ( config.isEnabled( 'tools/migrate' ) ) {
+			const { migrationStatus, currentSelectedSiteId } = this.props;
+
+			if ( currentSelectedSiteId && migrationStatus === 'error' ) {
+				wpcom.undocumented().resetMigration( currentSelectedSiteId );
```

Comment:
Does this need any sort of error handling? Also, such a shame to have to import the entire `lib/wpcom` library just for this.  We're trying to reduce the places where we're doing that because it's quite big. And the logged in masterbar is displayed literally on every Calypso page. Could we define this as a data layer handler? Or perhaps `requestHttpData()`?

Reply:
Thanks for the suggestion! Updated the code and now it should be lighter :) Not sure how to trigger the matticbot check again though :(

Reply:
It would usually retrigger at the end of the ICFY job. You can see that it automatically updates the matticbot message.